Database exception using DbSession

### What steps will reproduce the problem?
When I'm trying to login on the website, from time to time I'm receiving an error message.
After reloading of the page everything is ok.
`YII_DEBUG `is enabled, `YII_ENV_DEV` is set to true.

Maybe this issue is linked with [this one](https://github.com/yiisoft/yii2/pull/1812)

[Stackoverflow question](http://stackoverflow.com/questions/41158812/yii2-dbsession-exception)
### What do you get instead?

> Integrity constraint violation – yii\db\IntegrityException
> 
> SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY'
> The SQL being executed was: UPDATE `session` SET `id`=0 WHERE `id`='3u5r7gq6pqh8s0gjrusfqpmni7'
> Error Info: Array
> (
>     [0] => 23000
>     [1] => 1062
>     [2] => Duplicate entry '0' for key 'PRIMARY'
> )
> ↵
> Caused by: PDOException
> 
> SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '0' for key 'PRIMARY'
> 
> in /var/www/develop/vendor/yiisoft/yii2/db/Command.php at line 844

### Additional info

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.10
| PHP version      | 7.0
| Operating system | Ubuntu 14.04

--- a/framework/CHANGELOG.md
+++ b/framework/CHANGELOG.md
@@ -41,6 +41,7 @@ Yii Framework 2 Change Log
 - Bug #13159: Fixed `destroy` method in `yii.captcha.js` which did not work as expected (arogachev)
 - Bug #13198: Fixed order of checks in `yii\validators\IpValidator` that sometimes caused wrong error message (silverfire)
 - Bug #13200: Creating URLs for routes specified in `yii\rest\UrlRule::$extraPatterns` did not work if no HTTP verb was specified (cebe)
+- Bug #13212: Fixed `DbSession::regenerateID()` failure when `session_regenerate_id()` fails (andrewnester)
 - Bug #13229: Fix fetching schema information for `pgsql` when `PDO::ATTR_CASE` is set (klimov-paul)
 - Bug #13231: Fixed `destroy` method in `yii.gridView.js` which did not work as expected (arogachev)
 - Bug #13232: Event handlers were not detached with changed selector in `yii.gridView.js` (arogachev)

--- a/framework/web/DbSession.php
+++ b/framework/web/DbSession.php
@@ -102,6 +102,11 @@ class DbSession extends MultiFieldSession
 
         parent::regenerateID(false);
         $newID = session_id();
+        // if session id regeneration failed, no need to create/update it.
+        if (empty($newID)) {
+            Yii::warning('Failed to generate new session ID', __METHOD__);
+            return;
+        }
 
         $query = new Query();
         $row = $query->from($this->sessionTable)

